File: src/#20/solution.js
var isValid = function (s) {
  let stack = []
  if (s.length === 1) {
    return false
  }
  s.split('')
  for (let i = 0; i < s.length; i++) {
    let c = s[i]
    if (i === 0) {
      if (c === ')' || c === '}' || c === ']') {
        return false
      }
    }
    if (c === '(' || c === '{' || c === '[') {
      stack.push(c)
    } else if (c === ']') {
      if (stack.length > 0 && stack[stack.length - 1] === '[') {
        stack.pop()
      } else {
        return false
      }
    } else if (c === '}') {
      if (stack.length > 0 && stack[stack.length - 1] === '{') {
        stack.pop()
      } else {
        return false
      }
    } else if (c === ')') {
      if (stack.length > 0 && stack[stack.length - 1] === '(') {
        stack.pop()
      } else {
        return false
      }
    }
  }

  return stack.length === 0
}
console.log(isValid('(])'))

File: src/#20/solution.go
package solution

func isValid(s string) bool {
	stack := []byte{}
	if len(s) == 1 {
		return false
	}
	for i, c := range []byte(s) {
		if i == 0 {
			if c == '}' || c == ')' || c == ']' {
				return false
			}
		}
		stackLen := len(stack)
		if c == '[' || c == '{' || c == '(' {
			stack = append(stack, c)
		} else if c == ']' {
			if stackLen > 0 && stack[stackLen-1] == '[' {
				stack = stack[:len(stack)-1]
			} else {
				return false
			}
		} else if c == '}' {
			if stackLen > 0 && stack[stackLen-1] == '{' {
				stack = stack[:len(stack)-1]
			} else {
				return false
			}
		} else if c == ')' {
			if stackLen > 0 && stack[stackLen-1] == '(' {
				stack = stack[:len(stack)-1]
			} else {
				return false
			}
		}
	}
	return len(stack) == 0
}
